app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -38,13 +38,26 @@ def create_stripe_setup_intent
 
     delegate :payment_detail, to: :account
 
-    def customer
+    def customer_id
+      payment_detail.credit_card_auth_code
+    end
+
+    def find_or_create_customer
+      customer = Stripe::Customer.retrieve(customer_id, api_key) if customer_id
+      customer.try(:id) ? customer : create_customer
```

Comment:
Should we contempt the recently deleted case?
```suggestion
      customer.try(:id) && !customer.try(:deleted) ? customer : create_customer
```

Reply:
IMO already the `.try(:id)` can be overkill and could be simplified to:
```ruby
customer_id.present? ? Stripe::Customer.retrieve(customer_id, api_key) : create_customer
```
Because `Stripe::Customer.retrieve(customer_id, api_key)` either works and returns the customer with ID and everything, or just raises an error, but it is impossible to return any object without any ID.


That being said, the `deleted` check makes perfect sense to me 😉 thanks for the catch 👍🏼